tasks/oss.yml

```diff
@@ -28,20 +28,29 @@
 
 - name: Prepare storage devices
   block:
+    - name: Stat the OSS device
+      stat:
+        path: "{{ oss_dev }}"
+      register: oss_dev_stat
+    - name: Fail if OSS device does not exist
+      fail:
+        msg: OSS device {{ oss_dev }} does not exist
+      when: not oss_dev_stat.exists
     - name: Unmount storage device if beegfs_force_format is true
       mount:
-        src: "{{ oss_dev }}"
         path: "{{ oss_path }}"
         state: unmounted
       when: beegfs_force_format | bool
       notify: Restart BeeGFS storage service
     - name: Attempt to format if the device is not mounted or if beegfs_force_format is true
+      vars:
+        oss_dev_real: "{{ oss_dev_stat.lnk_source if oss_dev_stat.islnk else oss_dev_stat.path }}"
```

Comment:
Is this to support use case where oss_dev is a symlink?

Reply:
Yes. Devices in /dev/disk/by-id/<id> are symlinks to /dev/<dev>